plugins/node/opentelemetry-instrumentation-mysql/src/instrumentation.ts

```diff
@@ -216,7 +223,9 @@ export class MySQLInstrumentation extends InstrumentationBase<
         }
       }
       if (typeof cb === 'function') {
-        cb(...arguments);
+        context.with(activeContext, () => {
```

Comment:
`this` is not passed to the callback here.

I think following should work which additionally avoids the allocation of the arrow function:
```js
context.with(activeContext, cb, this, ...arguments)
```

Reply:
I prefer your version because its safer but shouldn't the correct `this` be inhered since its an arrow fn ?

Reply:
The `this` visible in the arrow function should be the correct one but it's not forwarded into `cb`. It would be needed use e.g. `Reflect.apply(cb, this, arguments)` to forward it.
But as mentioned above it's not needed to allocate the arrow function at all.

Reply:
@Flarna thanks, I've made changes to use `context.with` without the arrow function instead. I had to make some additional changes around types to make the compiler happy though.